test/Constraints/overload.swift

```diff
@@ -290,9 +290,8 @@ func rdar97396399() {
 func f63834(int: Int, string: String) {} // expected-note 3{{found candidate with type '(Int, String) -> ()'}}
 func f63834(int: Int, string: Bool) {} // expected-note 3{{found candidate with type '(Int, Bool) -> ()'}}
 
-// FIXME: Note should state that candidate has 2 arguments but 1 is expected.
-func f63834_1(int: Int, string: Bool) {} // expected-note{{candidate '(Int) -> Void' requires 1 argument, but 2 were provided}}
-func f63834_1(int: Int, string: String) {} // expected-note{{candidate '(Int) -> Void' requires 1 argument, but 2 were provided}}
+func f63834_1(int: Int, string: Bool) {} // expected-note{{contextual type '(Int) -> Void' requires 1 argument, but 2 were provided by candidate of type '(Int, Bool) -> ()'}}
+func f63834_1(int: Int, string: String) {} // expected-note{{contextual type '(Int) -> Void' requires 1 argument, but 2 were provided by candidate of type '(Int, String) -> ()'}}
```

Comment:
* IMO the note should start with a description of the candidate, because that is what it point to.
* The candidate "requires" rather than "provides".

But I’m not sure if it’s safe to modify this diagnostic any further without the test coverage.

Reply:
> * The candidate "requires" rather than "provides".

My understanding is that since there is a candidate is trying to match a contextual type, I read as `context requires one thing but candidate provides another`. I think candidate "requires" makes sense in a call situation.

> But I’m not sure if it’s safe to modify this diagnostic any further without the test coverage.

I think is fine to change at some extent since we are adding a test case. Although I wouldn't like to touch closure one given I still have no idea on how to repro this note with this scenario.

Reply:
> * IMO the note should start with a description of the candidate, because that is what it point to.

What about `candidate '(Int, String) -> ()' has 2 arguments but context '(Int) -> Void' requires 1`

Reply:
> But I’m not sure if it’s safe to modify this diagnostic any further without the test coverage.

Separated the closure one to leave it untouched for now.

Reply:
> What about `candidate '(Int, String) -> ()' has 2 arguments, but context '(Int) -> Void' requires 1`?

To be nitpicky, candidates don't have arguments, they have parameters. Maybe it would be better worded as something like `candidate '(Int, String) -> ()' has 2 parameters, but context '(Int) -> Void' has 1`?